pkg/kubectl/cmd/macro/macro.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package macro
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+// createOptions is a place-holder for options used by createInternal across all macro commands
+// TODO: make public this may be useful to downstream components that add own macro commands
+type createOptions struct {
+	// the factory object
+	f *cmdutil.Factory
+	// the command output writer
+	cmdOut io.Writer
+	// the command being executed
+	cmd *cobra.Command
```

Comment:
i think if i renamed this struct it would have been clearer.  it's just a holder for all things that get passed into the utility that each macro shares so we avoid having a ton of duplicate code for each macro.